packages/jest-jasmine2/src/index.js

```diff
@@ -69,7 +69,32 @@ function jasmine2(config, environment, moduleLoader, testPath) {
       moduleLoader.requireModule(null, config.setupTestFrameworkScriptFile);
     }
   });
+
+  const iterableEquality = (left, right) => {
+    let result; // result needs to be `undefined` if not an iterator
+    if (left && right && left[Symbol.iterator] && right[Symbol.iterator]) {
+      const leftIterator = left[Symbol.iterator]();
+      const rightIterator = right[Symbol.iterator]();
+      let nextLeft = leftIterator.next();
+      let nextRight = rightIterator.next();
+
+      result = true;
+      while (result && !nextLeft.done) {
+        result = jasmine.matchersUtil.equals(nextLeft, nextRight);
```

Comment:
I would also change the iteration to not use the `result` variable at all:

```js
while (!aItem.done) {
  if (!jasmine.matchersUtil.equals(aItem.value, bItem.value)) {
    return false;
  }
}

if (!bItem.done) {
  return false;
}
```

This way we avoid mutating variables completely :)

Also it seems like you are passing the iterator return value to the comparator instead of the `.value` field. It works because that is just an object so it does a deep-compare.